packages/ui-components/src/components/Matrix/MatrixCell.tsx

```diff
@@ -89,16 +85,24 @@ export const MatrixCell = ({ value, rowTitle, isCategory, colKey, onClick }: Mat
     getIsUsingDots(presentationOptionsForCell) &&
     checkIfApplyDotStyle(presentationOptionsForCell as ConditionalPresentationOptions, colIndex);
   const presentation = getPresentationOption(presentationOptionsForCell, value);
-  const displayValue = isDots ? (
-    <Dot
-      $color={presentation?.color}
-      aria-label={`${presentation?.description ? `${presentation.description}: ` : ''}${
-        value || 'No value'
-      }`}
-    />
-  ) : (
-    value
-  );
+
+  const getDisplayValue = () => {
+    if (isDots) {
+      return (
+        <Dot
+          $color={presentation?.color}
+          aria-label={`${presentation?.description ? `${presentation.description}: ` : ''}${
+            value || 'No value'
```

Comment:
Minor, but I personally found this expression a bit of a mind-bender

```suggestion
          aria-label={!!presentation?.description ? `${presentation.description}: ${value}` : 'No value'}
```